app/grandchallenge/components/serializers.py

```diff
@@ -26,32 +31,139 @@ class SimpleImageSerializer(serializers.ModelSerializer):
     # name is needed
     class Meta:
         model = Image
-        fields = (
-            "pk",
-            "name",
-        )
+        fields = ("pk", "name")
 
 
-class ComponentInterfaceValueSerializer(serializers.ModelSerializer):
-    # Serializes images in place rather than with hyperlinks for internal usage
-    image = SimpleImageSerializer()
-    interface = ComponentInterfaceSerializer()
+class ComponentInterfaceValuePostSerializer(serializers.ModelSerializer):
+    """
+    Serializes images with hyperlinks for external usage
+    Expects interface_slug, to allow creating a ComponentInterfaceValue with an
+    existing ComponentInterface
+    """
+
+    image = serializers.HyperlinkedRelatedField(
+        queryset=Image.objects.all(),
+        view_name="api:image-detail",
+        required=False,
+    )
+
+    interface_slug = serializers.CharField(write_only=True)
+    interface = ComponentInterfaceSerializer(read_only=True)
```

Comment:
Rather than having the object change between post and read we could replace both `interface_slug` and `interface` with:

```
interface = SlugRelatedField(queryset=...,slug_field="slug")
```

What do you think? Would this be too inconsistent with the pks and hyperlinks used elsewhere? I think it would save some of the things happening in L71-L82 in this file.

Reply:
Surprised that this works, thought I would get an "interface already exists" warning. Much better!